XHR error events should return 0 for loaded & total

cdumez opened the pull request:
#### 2040449337eec2b7d422ecf01b6082b1a0aaf9ea
<pre>
XHR error events should return 0 for loaded &amp; total
<a href="https://bugs.webkit.org/show_bug.cgi?id=243085">https://bugs.webkit.org/show_bug.cgi?id=243085</a>

Reviewed by Darin Adler.

XHR error events should return 0 for loaded &amp; total:
- <a href="https://xhr.spec.whatwg.org/#handle-errors">https://xhr.spec.whatwg.org/#handle-errors</a> (steps 7 &amp; 8)

This behavior matches Blink &amp; Gecko.

* LayoutTests/imported/w3c/web-platform-tests/xhr/event-error.sub.any-expected.txt:
* Source/WebCore/xml/XMLHttpRequest.cpp:
(WebCore::XMLHttpRequest::dispatchErrorEvents):
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.cpp:
(WebCore::XMLHttpRequestProgressEventThrottle::dispatchProgressEvent):
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.h:

Canonical link: <a href="https://commits.webkit.org/252743@main">https://commits.webkit.org/252743@main</a>
</pre>

darinadler commented on Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.h:
```diff
@@ -50,6 +50,7 @@ class XMLHttpRequestProgressEventThrottle {
     void updateProgress(bool isAsync, bool lengthComputable, unsigned long long loaded, unsigned long long total);
     void dispatchReadyStateChangeEvent(Event&, ProgressEventAction = DoNotFlushProgressEvent);
     void dispatchProgressEvent(const AtomString&);
+    void dispatchProgressEvent(const AtomString&, bool lengthComputable, unsigned long long loaded, unsigned long long total);
```
Another alternative would be to name this `dispatchProgressErrorEvent` and then not pass in the false and the zeros.

cdumez replied:
I went with your alternative proposal, I like it a little better.

cdumez committed:
XHR error events should return 0 for loaded & total
https://bugs.webkit.org/show_bug.cgi?id=243085

Reviewed by Darin Adler.

XHR error events should return 0 for loaded & total:
- https://xhr.spec.whatwg.org/#handle-errors (steps 7 & 8)

This behavior matches Blink & Gecko.

* LayoutTests/imported/w3c/web-platform-tests/xhr/event-error.sub.any-expected.txt:
* Source/WebCore/xml/XMLHttpRequest.cpp:
(WebCore::XMLHttpRequest::dispatchErrorEvents):
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.cpp:
(WebCore::XMLHttpRequestProgressEventThrottle::dispatchProgressEvent):
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.h:

Canonical link: https://commits.webkit.org/252743@main
